planner/core/task.go

```diff
@@ -557,28 +557,186 @@ func (p *PhysicalHashJoin) attach2Task(tasks ...task) task {
 	return task
 }
 
+// for different decimal scale and precision, tiflash uses different underlying type.
+// Here we check the scale and precision to decide whether conversion is a must.
+func needDecimalConvert(tp *types.FieldType, dec int, l int) bool {
```

Comment:
can here use the same date type `types.FieldType`? because `dec int, l int` belong to a `FieldType`.
Besides, thin function just uses the flen is necessary.

Reply:
ok